### PR TITLE
api cleanup

### PR DESCRIPTION
**Pull request by cvaroqui**

### CODE REVIEW

File: daemon/daemonauth/auth.go [hunk 1]
```diff
@@ -58,7 +61,13 @@ func MiddleWare(_ context.Context) func(http.Handler) http.Handler {
 			if err != nil {
 				log.Logger.Error().Err(err).Str("remote", r.RemoteAddr).Msg("auth")
 				code := http.StatusUnauthorized
-				http.Error(w, http.StatusText(code), code)
+				w.Header().Set("Content-Type", "application/json")
```
Comment (cgalibern) [inline daemon/daemonauth/auth.go:64]: can be replaced by WriteProblem

File: drivers/resdiskdrbd/main.go [hunk 2]
```diff
@@ -655,10 +655,20 @@ func (t T) sendConfigToNode(nodename string, allocationId uuid.UUID, b []byte) e
...
+	switch resp.StatusCode() {
+	case 400:
+		return errors.Errorf("%s", resp.JSON400)
+	case 401:
+		return errors.Errorf("%s", resp.JSON401)
+	case 403:
+		return errors.Errorf("%s", resp.JSON403)
+	case 500:
+		return errors.Errorf("%s", resp.JSON500)
+
```
Comment (cgalibern) [inline drivers/resdiskdrbd/main.go:671]: add default that returns error unexpected status code

File: daemon/daemonapi/get_networks.go [hunk 3]
```diff
@@ -11,9 +12,11 @@ import (
 
 // GetDaemonDNSDump returns the DNS zone content.
 func (a *DaemonApi) GetNetworks(w http.ResponseWriter, r *http.Request, params api.GetNetworksParams) {
+	WriteProblem(w, http.StatusInternalServerError, "Failed to allocate a new object.Node", "foo")
```
Comment (cgalibern) [inline daemon/daemonapi/get_networks.go:15]: TODO, remove this

File: daemon/api/main.go [hunk 4]
```diff
@@ -3,3 +3,20 @@
...
 package api
+
+import "fmt"
+
+func (t MonitorUpdateQueued) String() (out string) {
+	return fmt.Sprint(t.OrchestrationId)
+}
+
+func (t Problem) String() (out string) {
+	if t.Status != 200 {
```
Comment (cgalibern) [inline daemon/api/main.go:14]: Relax this to  < 300 ?

File: core/objectaction/object.go [hunk 5]
```diff
@@ -399,36 +400,54 @@ func (t T) DoAsync() error {
...
+			resp, e := c.PostObjectSwitchToWithResponse(ctx, params)
+			if e != nil {
 				err = e
-			} else if resp.StatusCode() != http.StatusOK {
-				err = errors.Errorf("unexpected post object switch status %s", resp.Status())
-			} else {
+			}
+			switch resp.StatusCode() {
+			case http.StatusOK:
 				b = resp.Body
```
Comment (cgalibern) [inline core/objectaction/object.go:409]: can we use JSON 200 here ?

File: core/nodeaction/node.go [hunk 6]
```diff
@@ -264,24 +264,29 @@ func (t T) DoAsync() error {
...
+			Colorize:      rawconfig.Colorize,
+		}.Print()
+	case 400:
+		return errors.Errorf("%s", resp.JSON400)
+	case 401:
+		return errors.Errorf("%s", resp.JSON403)
+	case 403:
+		return errors.Errorf("%s", resp.JSON401)
+	case 500:
+		return errors.Errorf("%s", resp.JSON500)
```
Comment (cgalibern) [inline core/nodeaction/node.go:288]: add default ?

**Review by cgalibern — CHANGES_REQUESTED** [review]
:+1 with some comments

File: daemon/daemonauth/auth.go (hunk 1 above)
Comment (cgalibern) [reply daemon/daemonauth/auth.go:64]: forget this

File: daemon/api/main.go [hunk 7]
```diff
@@ -3,3 +3,20 @@
...
+		out += fmt.Sprintf("%s ", t.Status)
```
Comment (cgalibern) [inline daemon/api/main.go:15]: %s -> %d

File: daemon/imon/orchestration.go [hunk 8]
```diff
@@ -110,14 +111,14 @@ func (o *imon) endOrchestration() {
...
-	o.state.OrchestrationId = ""
+	o.state.OrchestrationId = uuid.UUID{}
 }
 
 // isConvergedOrchestrationReached returns true instance orchestration is reached
 // for all object instances (OrchestrationId == "" for all instance monitor cache).
 func (o *imon) isConvergedOrchestrationReached() bool {
 	for nodename, oImon := range o.instMonitor {
-		if oImon.OrchestrationId != "" {
+		if oImon.OrchestrationId.String() != "" {
```
Comment (cgalibern) [inline daemon/imon/orchestration.go:121]: use zero value here

File: daemon/imon/orchestration.go [hunk 9]
```diff
@@ -92,16 +93,16 @@ func (o *imon) endOrchestration() {
 	o.state.GlobalExpectOptions = nil
 	o.clearPending()
 	o.updateIfChange()
-	if o.acceptedOrchestrationId != "" {
+	if o.acceptedOrchestrationId.String() != "" {
```
Comment (cgalibern) [inline daemon/imon/orchestration.go:96]: use zero value here or != "00000000-0000-0000-0000-000000000000"

File: daemon/api/main.go (hunk 4 above)
Comment (cvaroqui) [reply daemon/api/main.go:14]: Better to display the code if not 200 to avoid confusion

**Review by cgalibern — APPROVED** [review]

**Pull request merged**